app/serializers/disability_claim_detail_serializer.rb

```diff
@@ -51,30 +51,44 @@ def create_event_from_string_date(type, *from_keys)
   ).freeze
 
   def create_events_for_tracked_items
-    events = []
-    TRACKED_ITEM_FIELDS.each do |field|
-      list_objects_with_key(['claimTrackedItems', field], ['openedDate']).each do |obj|
-        date = obj['openedDate'] || obj['receivedDate']
-        events << {
-          type: field.snakecase,
-          date: Date.strptime(date, '%m/%d/%Y'),
-          description: obj['description'],
-          display_name: obj['displayedName'],
-          overdue: obj['overdue'],
-          tracked_item_id: obj['trackedItemId'],
-          tracked_item_status: obj['trackedItemStatus']
-        } if date
+    TRACKED_ITEM_FIELDS.map do |field|
+      sub_objects_of('claimTrackedItems', field).map do |obj|
+        create_tracked_item_event(field.snakecase, obj)
       end
-    end
-    events
+    end.flatten
+  end
+
+  def create_tracked_item_event(type, obj)
+    event = {
+      type: type,
+      tracked_item_id: obj['trackedItemId'],
+      description: obj['description'],
+      display_name: obj['displayedName'],
+      overdue: obj['overdue'],
+      status: obj['trackedItemStatus'],
+      uploaded: obj['uploaded'],
+      uploads_allowed: obj['uploadsAllowed'],
+      opened_date: date_or_nil_from(obj, 'openedDate'),
+      requested_date: date_or_nil_from(obj, 'requestedDate'),
+      received_date: date_or_nil_from(obj, 'receivedDate'),
+      closed_date: date_or_nil_from(obj, 'closedDate'),
+      suspense_date: date_or_nil_from(obj, 'suspenseDate')
+    }
+    event[:date] = [
+      event[:opened_date], event[:requested_date], event[:received_date],
```

Comment:
I didn't know about `slice`. That looks good. It also helps me stay under rubocops twenty line per method limit (which is why this is a separate method in the first place).

The order will determine where the item shows up in the chronological timeline. I.e. something might be scheduled to become invalid in a month, but that doesn't mean it should be in the timeline before something that just happened. OTOH, if something was received yesterday, that's more important than the fact that it was opened two weeks ago (which is wrong in the current commit).

I'll tweak this and then merge, so that Jeff and Mo can start using this data.